tests/adapters/structures/test_aiida.py

```diff
@@ -0,0 +1,89 @@
+# pylint: disable=import-error
+import json
+from pathlib import Path
+import re
+from typing import List
+
+import pytest
+
+from .utils import get_min_ver
+
+min_ver = get_min_ver("aiida-core")
+aiida = pytest.importorskip(
+    "aiida",
+    minversion=min_ver,
+    reason=f"aiida-core must be installed with minimum version {min_ver} for these tests to"
+    " be able to run",
+)
+
+from aiida import load_profile
+
+load_profile()
+
+from aiida.orm import StructureData
+
+from optimade.models.structures import Periodicity
+
+from optimade.adapters import Structure
+from optimade.adapters.exceptions import ConversionError
+from optimade.adapters.structures.aiida import get_aiida_structure_data
+
+
+with open(Path(__file__).parent.joinpath("raw_test_structures.json"), "r") as raw_data:
+    RAW_STRUCTURES: List[dict] = json.load(raw_data)
+
+
+def test_successful_conversion():
+    """Make sure its possible to convert"""
+    for structure in RAW_STRUCTURES:
+        assert isinstance(get_aiida_structure_data(Structure(structure)), StructureData)
+
+
+def test_null_positions():
+    """Make sure null positions are handled"""
+    structure = Structure(RAW_STRUCTURES[0])
+    structure.attributes.cartesian_site_positions[0] = (None,) * 3
+    with pytest.raises(ConversionError) as exc:
+        get_aiida_structure_data(structure)
+    assert "AiiDA cannot be used to convert structures with unknown positions." == str(
+        exc.value
+    )
+
+
+def test_null_lattice_vectors():
```

Comment:
Although, how would you ever receive such a `lattice_vector` value? That's not OPTIMADE compliant as far as I know? It should always be a 3 long list of 3 long lists, i.e., `((None, None, None), (1, 1, 1), (1, 1, 1))`? Do you think we should try and treat cases that the OPTIMADE spec says are not allowed as well? Just curious, as I'm having a bit of trouble deciding how far these adapters should go in error handling.

Reply:
Ah I must have mixed up the lattice vector discussion in my head, for some reason I thought we were allowing `None` instead of `(None, None, None)`. Feel free to resolve any other comments on this...